packages/core/src/checkout-buttons/checkout-headless-button-initializer.ts

```diff
@@ -0,0 +1,152 @@
+            this._state = createCheckoutButtonSelectors(state);
+        });
+    }
+
+    /**
+     * Returns a snapshot of the current state.
+     *
+     * The method returns a new instance every time there is a change in the
+     * state. You can query the state by calling any of its getter methods.
+     *
+     * ```js
+     * const state = service.getState();
+     *
+     * console.log(state.errors.getInitializeButtonError());
+     * console.log(state.statuses.isInitializingButton());
+     * ```
+     *
+     * @returns The current customer's checkout state
+     */
+    getState(): CheckoutButtonSelectors {
+        return this._state;
+    }
+
+    /**
+     * Subscribes to any changes to the current state.
+     *
+     * The method registers a callback function and executes it every time there
+     * is a change in the current state.
+     *
+     * ```js
+     * service.subscribe(state => {
+     *     console.log(state.statuses.isInitializingButton());
+     * });
+     * ```
+     *
+     * The method can be configured to notify subscribers only regarding
+     * relevant changes, by providing a filter function.
+     *
+     * ```js
+     * const filter = state => state.errors.getInitializeButtonError();
+     *
+     * // Only trigger the subscriber when the cart changes.
+     * service.subscribe(state => {
+     *     console.log(state.errors.getInitializeButtonError())
+     * }, filter);
+     * ```
+     *
+     * @param subscriber - The function to subscribe to state changes.
+     * @param filters - One or more functions to filter out irrelevant state
+     * changes. If more than one function is provided, the subscriber will only
+     * be triggered if all conditions are met.
+     * @returns A function, if called, will unsubscribe the subscriber.
+     */
+    subscribe(
+        subscriber: (state: CheckoutButtonSelectors) => void,
+        ...filters: Array<(state: CheckoutButtonSelectors) => any>
+    ): () => void {
+        return this._store.subscribe(
+            () => subscriber(this.getState()),
+            (state) => state.checkoutButton.getState(),
+            ...filters.map(
+                (filter) => (state: InternalCheckoutSelectors) =>
+                    filter(createCheckoutButtonSelectors(state)),
+            ),
+        );
+    }
+
+    /**
+     * Initializes the headless wallet button of a payment method.
+     *
+     * When the headless wallet button is initialized, it will be inserted into the DOM,
+     * ready to be interacted with by the customer.
+     *
+     * ```js
+     * initializer.initializeHeadlessButton({
+     *     methodId: 'braintreepaypal',
+     *     containerId: 'headlessWalletButton',
+     *     braintreepaypal: {
+     *     },
+     * });
+     * ```
+     *
+     * @param options - Options for initializing the checkout button.
+     * @returns A promise that resolves to the current state.
+     */
+    initializeHeadlessButton(
```

Comment:
Can you explain why we don't call this method, `intialize`?

Reply:
What do you mean we do not call this method?

Reply:
In other initializers, we refer to this method as initialize. I was wondering if there is a specific reason we didn't follow this pattern.